src/main/webapp/app/tasks/filter-pipe/filter-task.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Task } from '../task';
+
+@Pipe({
+  name: 'filterTask',
+  pure: false
+})
+export class FilterTaskPipe implements PipeTransform {
+
+  transform(tasks: Task[], text: string): Task[] {
+    if (text) {
+      if (text.includes('AND')) {
+        const splittedText = text.split('AND');
+        return tasks.filter(task => splittedText.every(part => task.name.toLowerCase().includes(part.trim().toLowerCase())));
```

Comment:
should extract `task.name.toLowerCase().includes(part.trim().toLowerCase()` to a single method because it is re-used three times

Reply:
Right, fixed